### DISCUSSION TITLE
Is there any way to reduce an image's colors to a specified palette using the C++ API?

### DISCUSSION THREAD

**Discussion by MrAnderson1971** [ImageMagick/ImageMagick#6374] [Help] [answered]
### ImageMagick version

7.1.1-Q8

### Operating system, version and so on

Windows 10 (x86)

### Description

Takes in an image and color palette, outputs a new image such that every pixel in the old image is replaced with the nearest pixel in the color palette.

**Comment by fmw42** [comment, marked as answer]
See -remap at https://imagemagick.org/Usage/quantize/#web_safe.  The color palette in this case is just an image with the desired colors. You can make that by appending 1x1 sized images of the desired colors or extracting them as the unique colors in some other image.
